### PR TITLE
Add getEntityKeyIds to DataIntegrationApi

### PR DESCRIPTION
**Pull request by bergieboy**

### CODE REVIEW

File: src/constants/UrlConstants.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ export const EDM_PATH :string = 'edm';
 export const EMAIL_DOMAINS_PATH :string = 'email-domains';
 export const EMAIL_PATH :string = 'email';
 export const ENTITY_SETS_PATH :string = 'entity-sets';
+export const ENTITY_KEY_IDS_PATH :string = 'entityKeyIds';
```
Comment (UnsungHero97) [inline src/constants/UrlConstants.js:32]: Please alphabetize.

File: src/api/DataIntegrationApi.js [hunk 2]
```diff
@@ -107,3 +108,39 @@ export function createEntityAndAssociationData(bulkDataCreation :Object) :Promis
       return Promise.reject(error);
     });
 }
+/**
+ * `POST /integration`
```
Comment (UnsungHero97) [inline src/api/DataIntegrationApi.js]: This should be `POST /integration/entityKeyIds`.

File: src/api/DataIntegrationApi.test.js [hunk 3]
```diff
@@ -121,5 +130,16 @@ function createEntityAndAssociationData() {
...
+
+function getEntityKeyIds() {
+  describe('getEntityKeyIds()', () => {
+
+    const apiToTest = DataIntegrationApi.getEntityKeyIds;
 
+    const validParams = MOCK_ENTITY_KEYS;
+
+    testApiShouldReturnPromise(apiToTest, validParams);
+    testApiShouldUseCorrectAxiosInstance(apiToTest, validParams, DATA_INTEGRATION_API);
```
Comment (UnsungHero97) [inline src/api/DataIntegrationApi.test.js:151]: You're missing these:

```
    testApiShouldNotThrowOnInvalidParameters(apiToTest, validParams, invalidParams);
    testApiShouldRejectOnInvalidParameters(apiToTest, validParams, invalidParams);
    testApiShouldSendCorrectHttpRequest(apiToTest, validParams, axiosParams, 'post');
```

File: src/api/DataIntegrationApi.js [hunk 4]
```diff
@@ -107,3 +108,39 @@ export function createEntityAndAssociationData(bulkDataCreation :Object) :Promis
...
+ * @return {Promise} - a Promise that resolves with a set of the entityKeyIds that were created for the given entityIds
+ *
+ * @example
+ *
+ * entityKey_1 = { entitySetId: "entitySetId_1", entityId: "entityId_1" };
+ * entityKey_2 = { entitySetId: "entitySetId_2", entityId: "entityId_2" };
+ *
+ * DataIntegrationApi.getEntityKeyIds({ entityIds: [entityKey_1, entityKey_2, ...] });
+ */
+export function getEntityKeyIds(entityKeys :string[]) :Promise<*> {
```
Comment (UnsungHero97) [inline src/api/DataIntegrationApi.js]: `entityKeys` is not a string array. This should be `:Object[]`, or if you want to be fancy...

```
type EntityKey = {|
  entityId :string;
  entitySetId :UUID;
|};
export function getEntityKeyIds(entityKeys :EntityKey[]) :Promise<*> {
  ...
}
```

File: src/api/DataIntegrationApi.js [hunk 5]
```diff
@@ -107,3 +108,39 @@ export function createEntityAndAssociationData(bulkDataCreation :Object) :Promis
...
+
+  let errorMsg = '';
+  console.log(entityKeys);
+
+  if (isUndefined(entityKeys) || !isNonEmptyArray(entityKeys)) {
+    errorMsg = 'invalid parameter: entityIdsCreation must be a non-empty set of entityIds';
```
Comment (UnsungHero97) [inline src/api/DataIntegrationApi.js]: What is `entityIdsCreation`? I think your error message needs to be...

```
errorMsg = 'invalid parameter: entityKeys must be a non-empty set of EntityKeys';
```

File: src/api/DataIntegrationApi.js [hunk 6]
```diff
@@ -107,3 +108,39 @@ export function createEntityAndAssociationData(bulkDataCreation :Object) :Promis
...
+ * @memberof lattice.DataIntegrationApi
+ * @param {Object} entityKeys
+ * @return {Promise} - a Promise that resolves with a set of the entityKeyIds that were created for the given entityIds
+ *
+ * @example
+ *
+ * entityKey_1 = { entitySetId: "entitySetId_1", entityId: "entityId_1" };
+ * entityKey_2 = { entitySetId: "entitySetId_2", entityId: "entityId_2" };
+ *
+ * DataIntegrationApi.getEntityKeyIds({ entityIds: [entityKey_1, entityKey_2, ...] });
```
Comment (UnsungHero97) [inline src/api/DataIntegrationApi.js]: This doesn't match the expected parameter, should be `getEntityKeyIds([entityKey_1, entityKey_2])`.

File: src/api/DataIntegrationApi.js [hunk 7]
```diff
@@ -107,3 +108,39 @@ export function createEntityAndAssociationData(bulkDataCreation :Object) :Promis
...
+ */
+export function getEntityKeyIds(entityKeys :string[]) :Promise<*> {
+
+  let errorMsg = '';
+  console.log(entityKeys);
+
+  if (isUndefined(entityKeys) || !isNonEmptyArray(entityKeys)) {
```
Comment (UnsungHero97) [inline src/api/DataIntegrationApi.js]: You don't need `isUndefined()`, `!isNonEmptyArray()` is enough.

File: src/api/DataIntegrationApi.js [hunk 8]
```diff
@@ -107,3 +108,39 @@ export function createEntityAndAssociationData(bulkDataCreation :Object) :Promis
...
+ *
+ * entityKey_1 = { entitySetId: "entitySetId_1", entityId: "entityId_1" };
+ * entityKey_2 = { entitySetId: "entitySetId_2", entityId: "entityId_2" };
+ *
+ * DataIntegrationApi.getEntityKeyIds({ entityIds: [entityKey_1, entityKey_2, ...] });
+ */
+export function getEntityKeyIds(entityKeys :string[]) :Promise<*> {
+
+  let errorMsg = '';
+  console.log(entityKeys);
```
Comment (UnsungHero97) [inline src/api/DataIntegrationApi.js]: No `console.log` please.

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

File: src/api/DataIntegrationApi.js [hunk 9]
```diff
@@ -107,3 +119,55 @@ export function createEntityAndAssociationData(bulkDataCreation :Object) :Promis
...
+ * @return {Promise} - a Promise that resolves with a set of the entityKeyIds that were created for the given entityIds
+ *
+ * @example
+ *
+ * entityKey_1 = { entitySetId: "entitySetId_1", entityId: "entityId_1" };
+ * entityKey_2 = { entitySetId: "entitySetId_2", entityId: "entityId_2" };
+ *
+ * DataIntegrationApi.getEntityKeyIds([entityKey_1, entityKey_2, ...]);
+ */
+export function getEntityKeyIds(entityKeys :EntityKey) :Promise<*> {
```
Comment (UnsungHero97) [inline src/api/DataIntegrationApi.js]: This should be `entityKeys :EntityKey[]` since it's an array.

File: src/api/DataIntegrationApi.js [hunk 10]
```diff
@@ -107,3 +119,55 @@ export function createEntityAndAssociationData(bulkDataCreation :Object) :Promis
...
+
+  let errorMsg = '';
+
+  if (!isNonEmptyArray(entityKeys)) {
+    errorMsg = 'invalid parameter: entityIdsCreation must be a non-empty set of entityKeys';
```
Comment (UnsungHero97) [inline src/api/DataIntegrationApi.js]: What is `entityIdsCreation`?

File: src/api/DataIntegrationApi.test.js [hunk 11]
```diff
@@ -61,6 +65,19 @@ const MOCK_ASSOCIATIATIONS = [
   }
 ];
 
+const MOCK_ENTITY_KEYS = [
+  { entitySetId: genRandomUUID(), entityId: genRandomString() },
+  { entitySetId: genRandomUUID(), entityId: genRandomString() },
+  { entitySetId: genRandomUUID(), entityId: genRandomString() },
+  { entitySetId: genRandomUUID(), entityId: genRandomString() }
+];
+const INVALID_MOCK_ENTITY_KEYS = [
```
Comment (UnsungHero97) [inline src/api/DataIntegrationApi.test.js]: You don't necessarily need this.

File: src/api/DataIntegrationApi.test.js [hunk 12]
```diff
@@ -121,5 +139,23 @@ function createEntityAndAssociationData() {
...
+}
+
+function getEntityKeyIds() {
+  describe('getEntityKeyIds()', () => {
+
+    const apiToTest = DataIntegrationApi.getEntityKeyIds;
 
+    const validParams = [MOCK_ENTITY_KEYS];
+
+    const invalidParams = [INVALID_MOCK_ENTITY_KEYS];
```
Comment (UnsungHero97) [inline src/api/DataIntegrationApi.test.js]: Due to the way the test helper function is written, using `INVALID_PARAMS` instead would cover more edge cases.

```
import { INVALID_PARAMS } from '../utils/testing/Invalid';
...
const invalidParams = [INVALID_PARAMS];
```

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

**Review by UnsungHero97 — APPROVED** [review]
